main_test.go

```diff
@@ -338,24 +339,53 @@ var _ = Describe("github-review-helper", func() {
 							})
 
 							Context("with head branch having changed", func() {
-								BeforeEach(func() {
+								mockMergeFailWithConflict := func() *mock.Call {
 									additionalCommitMessage := ""
 									resp := &http.Response{
 										StatusCode: http.StatusConflict,
 									}
-									pullRequests.
+									return pullRequests.
 										On("Merge", repositoryOwner, repositoryName, issueNumber, additionalCommitMessage).
 										Return(nil, &github.Response{
 											Response: resp,
 										}, &github.ErrorResponse{
 											Response: resp,
 											Message:  "Head branch was modified. Review and try the merge again.",
 										})
+								}
+
+								Context("every time", func() {
+									BeforeEach(func() {
+										mockMergeFailWithConflict()
+									})
+
+									It("reties 3 times and fails with a gateway error", func() {
```

Comment:
Made nicer with this: https://github.com/salemove/github-review-helper/pull/15/commits/78324fae250dcce115cec3835ce635364205415d